getTranslatedShaderSourceANGLE implemented

madrazo opened the pull request:
Updated from https://github.com/haxenme/nme/pull/518/files

Tested with 

```
 var fragShader:String =
"precision mediump float;\n" +
"void main(){
    // Output color = red
    gl_FragColor = vec4(1.0,0.0,0.0,1.0);
}
";

        var vertShader:String =
"precision mediump float;\n" +
"// Input vertex data, different for all executions of this shader.
uniform vec3 vertexPosition_modelspace;
void main(){
    gl_Position.xyz = vertexPosition_modelspace;
    gl_Position.w = 1.0;
}
";

var shaderProgram = nme.gl.Utils.createProgram(vertShader, fragShader);
var hlslSource =
nme.gl.GL.getTranslatedShaderSourceANGLE(shaderProgram.getShaders()[1]);
trace( ((hlslSource.split("// COMPILER INPUT HLSL BEGIN")[1]).split("// COMPILER INPUT HLSL END"))[0] );
```

Output
```
// GLSL
//
// #define texture texture2D// precision mediump float;// void main(){//     // Output color = red //     gl_FragColor = vec4(1.0,0.0,0.0,1.0);// }

#ifdef ANGLE_ENABLE_LOOP_FLATTEN
#define LOOP [loop]
#define FLATTEN [flatten]
#else
#define LOOP
#define FLATTEN
#endif
// Varyings

static float4 gl_Color[1] =
{
    float4(0, 0, 0, 0)
};

cbuffer DriverConstants : register(b1)
{
};

#define GL_USES_FRAG_COLOR
void gl_main()
{
(gl_Color[0] = float4(1.0, 0.0, 0.0, 1.0));
}
;
struct PS_INPUT
{
    float4 dx_Position : SV_Position;
    float4 gl_Position : TEXCOORD0;
};

struct PS_OUTPUT
{
    float4 gl_Color0 : SV_TARGET0;
};

PS_OUTPUT generateOutput()
{
    PS_OUTPUT output;
    output.gl_Color0 = gl_Color[0];
    return output;
}

PS_OUTPUT main(PS_INPUT input)
{

    gl_main();

    return generateOutput();
}
```

thomasuster commented:
@madrazo Hey there. I don't know GLSL. Why is this useful / what problem does it solve?

madrazo commented:
@thomasuster I used this as a tool for protoptyping postprocess shaders on GLSL ([nme-go](https://github.com/madrazo/nme-go)) and then used this function to translate it more quickly into HLSL (other engine that uses HLSL)

madrazo commented:
Tested this again and still works, but had to change the "split" strings
in HerokuShaders, Main.hx, line 184 add

```
      if (vs == null || fs == null) return;

      //test
   var hlslSource:String = nme.gl.GL.getTranslatedShaderSourceANGLE(fs);
   var hlslSource2:String =((hlslSource.split("// INITIAL HLSL BEGIN")[1]).split("// COMPILER INPUT HLSL END"))[0];
   trace( hlslSource2 );
```
